src/components/useWeatherData.js

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+
+// API call
+const api = {
+  key: "44082ee5fdfb89d4da8cad9697c38e7f",
+  url: "https://api.openweathermap.org/data/2.5/weather",
+};
+
+export function useWeatherData(cityName) {
+  const [weatherData, setWeatherData] = useState(null);
+
+  // fetch the weather data on mount and save to state
+  useMount(() => {
```

Comment:
Very cool to have a "init" function wrapping your useEffect, I might have to borrow this...